kornia/filters/kernels.py

```diff
@@ -616,13 +616,13 @@ def get_gaussian_kernel2d(
     KORNIA_CHECK_IS_TENSOR(sigma)
     KORNIA_CHECK_SHAPE(sigma, ["B", "2"])
 
-    ksize_x, ksize_y = _unpack_2d_ks(kernel_size)
-    sigma_x, sigma_y = sigma[:, 0, None], sigma[:, 1, None]
+    ksize_y, ksize_x = _unpack_2d_ks(kernel_size)
+    sigma_y, sigma_x = sigma[:, 0, None], sigma[:, 1, None]
 
-    kernel_x = get_gaussian_kernel1d(ksize_x, sigma_x, force_even, device=device, dtype=dtype)[..., None]
     kernel_y = get_gaussian_kernel1d(ksize_y, sigma_y, force_even, device=device, dtype=dtype)[..., None]
+    kernel_x = get_gaussian_kernel1d(ksize_x, sigma_x, force_even, device=device, dtype=dtype)[..., None]
 
-    return torch.matmul(kernel_x, kernel_y.transpose(2, 1))
+    return kernel_y * kernel_x.view(-1, 1, ksize_x)
```

Comment:
matmul ==  `@` ?

Reply:
Yes, @ is matmul. But in this case, we are doing [3x1] x [1x3] -> [3x3], which is equivalent to simple multiplication with broadcasting. The simplification is more evident in the 3D kernel case. The test cases pass so it should be correct.

Perhaps this change should be done in a separate PR?

Reply:
I see -- defenitely